extensions/eda/plugins/event_filter/insert_hosts_to_meta.py

```diff
@@ -32,20 +35,29 @@
 import dpath
 
 
+class PathNotExistError(Exception):
+    """Cannot find the path in the event."""
+
+
 def main(
     event: dict[str, Any],
     host_path: str | None = None,
     host_separator: str | None = None,
     path_separator: str = ".",
+    *,
+    raise_error: bool = False,
 ) -> dict[str, Any]:
     """Extract hosts from event data and insert into meta dict."""
     if not host_path:
         return event
 
     try:
         hosts = dpath.get(event, host_path, path_separator)
-    except KeyError:
+    except KeyError as error:
         # does not contain host
+        if raise_error:
```

Comment:
@bzwei Should we also log a warning message here irrespective of the raise_error setting?

Reply:
It may be normal for events not containing a host. This may generate a lot of noises in the log.

Reply:
Agree. Maybe INFO level.